### PR TITLE
feat: list by alternate timestamp such as changed

### PR DESCRIPTION
**Pull request by robinnedergard**
- [x] https://github.com/utily/transactly/pull/36

### CODE REVIEW

**Review by asnahassan — APPROVED** [review]

File: Backend.ts [hunk 1]
```diff
@@ -61,9 +62,16 @@ export class Backend<T> {
...
+						.map(
+							c =>
+								`a["value"]["${alternateDateTimeProperty ? alternateDateTimeProperty : "created"}"] ${c.operator} ${
+									c.name
+								}`
+						)
+						.join(" AND ")} `
 				: ""
-		}ORDER BY a["value"]["created"] DESC`
+		}ORDER BY a["value"]["${alternateDateTimeProperty ? alternateDateTimeProperty : "created"}"] DESC`
```
Comment (marcushanses) [inline Backend.ts:74]: alternateDateTimeProperty ?? "created"

File: Backend.ts [hunk 2]
```diff
@@ -61,9 +62,16 @@ export class Backend<T> {
 		]
 		const query = `SELECT * FROM ROOT a ${
 			parameters.length
-				? `WHERE ${parameters.map(c => `a["value"]["created"] ${c.operator} ${c.name}`).join(" AND ")} `
+				? `WHERE ${parameters
+						.map(
+							c =>
+								`a["value"]["${alternateDateTimeProperty ? alternateDateTimeProperty : "created"}"] ${c.operator} ${
```
Comment (marcushanses) [inline Backend.ts:68]: alternateDateTimeProperty ?? "created"

File: Backend.ts [hunk 3]
```diff
@@ -50,7 +50,8 @@ export class Backend<T> {
 		start?: isoly.Date | isoly.DateTime,
 		end?: isoly.Date | isoly.DateTime,
 		limit?: number,
-		continuation?: string
+		continuation?: string,
+		alternateDateTimeProperty?: string
```
Comment (marcushanses) [inline Backend.ts:54]: This name seems a bit convoluted, can't you name it something shorter?

Comment (Magnus93) [reply Backend.ts:54]: Alternatives: `timeProperty`, `time`, `property`, `indexBy`, `by`

You could also do
`property = "created"` on the argument. So you don't need the tertiary statement, this also does not make it an **alternative** property, just a property.

Comment (Magnus93) [reply Backend.ts:54]: You could also do
`alternateDateTimeProperty = "created"` on the argument. So you don't need the tertiary statement, this also does not make it an **alternative** property, just a property.

File: Backend.ts (hunk 1 above)
Comment (robinnedergard) [reply Backend.ts]: fixed

File: Backend.ts (hunk 2 above)
Comment (robinnedergard) [reply Backend.ts]: fixed

File: Backend.ts (hunk 3 above)
Comment (robinnedergard) [reply Backend.ts]: added default and shorted it to `dateTimeProperty`

**Review by marcushanses — APPROVED** [review]

**Pull request merged**